encoding.c

```diff
@@ -386,6 +393,25 @@ enc_register_at(struct enc_table *enc_table, int index, const char *name, rb_enc
     }
     encoding->name = name;
     encoding->ruby_encoding_index = index;
+    if (encoding->max_enc_len == 1 ||
+        STRCASECMP(name, "UTF-8") == 0 ||
+        STRCASECMP(name, "UTF8-MAC") == 0 ||
+        STRCASECMP(name, "UTF8-DoCoMo") == 0 ||
+        STRCASECMP(name, "UTF8-KDDI") == 0 ||
+        STRCASECMP(name, "UTF8-SoftBank") == 0 ||
+        STRCASECMP(name, "CP51932") == 0 ||
+        STRCASECMP(name, "Emacs-Mule") == 0 ||
+        STRCASECMP(name, "EUC-JIS-2004") == 0 ||
+        STRCASECMP(name, "EUC-JP") == 0 ||
+        STRCASECMP(name, "eucJP-ms") == 0 ||
+        STRCASECMP(name, "EUC-KR") == 0 ||
+        STRCASECMP(name, "EUC-TW") == 0 ||
+        STRCASECMP(name, "GB12345") == 0 ||
+        STRCASECMP(name, "GB2312") == 0 ||
+        STRCASECMP(name, "stateless-ISO-2022-JP") == 0 ||
+        STRCASECMP(name, "stateless-ISO-2022-JP-KDDI") == 0) {
+        encoding->flags |= ENC_ASCIISAFE_FLAG;
```

Comment:
I don't think this is the proper place to hardcode the list of ascii-safe encoding, but I really didn't know where else to put it.

Also, while ENC_DUMMY_FLAG is stored in `ruby_encoding_index`, I store the ENC_ASCIISAFE_FLAG bit in `flags` instead. It just seemed to make more sense?